src/main/java/duke/data/FileStorage.java

```diff
@@ -0,0 +1,69 @@
+package duke.data;
+
+import duke.entities.Task;
+import duke.entities.Tasklist;
+import duke.entities.Todo;
+
+import java.io.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class FileStorage implements IStorage {
+    private final String filePath;
+    private static final Pattern TASK_REGEX = Pattern.compile("^\\[(T|D|E)\\]\\[(x| )\\] (.*)(?: \\(.*: (.*)\\))?$");
+
+    public FileStorage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    @Override
+    public void write(Tasklist tasklist) {
+        try {
+            File file = new File(this.filePath);
+            file.createNewFile();
+        } catch (IOException e) {
+            System.out.println(String.format("Unable to create file: %s", this.filePath));
+        }
+
+        try {
+            FileWriter myWriter = new FileWriter(this.filePath);
+            for (Task task: tasklist) {

```

Comment:
```suggestion
            for (Task task : tasklist) {
```
I feel there should be an empty space in front of the colon here, but it depends on your IDE config. Feel free to close this if you disagree